buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -589,36 +589,26 @@ public synchronized String toString() {
             .append(q100)
             .append(StringUtil.NEWLINE)
             .append("tiny subpages:");
-        for (int i = 1; i < tinySubpagePools.length; i ++) {
-            PoolSubpage<T> head = tinySubpagePools[i];
-            if (head.next == head) {
-                continue;
-            }
-
-            buf.append(StringUtil.NEWLINE)
-               .append(i)
-               .append(": ");
-            PoolSubpage<T> s = head.next;
-            for (;;) {
-                buf.append(s);
-                s = s.next;
-                if (s == head) {
-                    break;
-                }
-            }
-        }
+        appendPoolSubPages(buf, tinySubpagePools);
         buf.append(StringUtil.NEWLINE)
            .append("small subpages:");
-        for (int i = 1; i < smallSubpagePools.length; i ++) {
-            PoolSubpage<T> head = smallSubpagePools[i];
+        appendPoolSubPages(buf, smallSubpagePools);
+        buf.append(StringUtil.NEWLINE);
+
+        return buf.toString();
+    }
+
+    private static void appendPoolSubPages(StringBuilder buf, PoolSubpage<?>[] subpages) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)